Refactor `updateSelectedSite` task - don't reset site data

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #9528 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR modifies `updateSelectedSite` in `AppInitializer`. As discussed (here: peaMlT-6L-p2), we observe that it makes more problems than benefits when resetting site data, while there are ongoing async jobs that use that data.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
1. Override `AppInitializer.SECONDS_BETWEEN_SITE_UPDATE` to 1
2. Move the app to the background and disable the WooCommerce plugin
3. Resume the app and observe it doesn't crash, just displays error snackbars

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Disable `updateSelectedSite` using feature flag

malinajirka commented:
Thanks for working on this @samiuelson!

I though we were discussing removing the following part of the code:
```
 if (it.model?.hasWooCommerce == false) {
                            // The previously selected site is not connected anymore, take the user to the site picker
                            WooLog.w(T.LOGIN, "Selected site no longer has WooCommerce")
                            selectedSite.reset()
                            restartMainActivity()
                        }
```

It seems this PR removes all updates of the site when the app comes from background. Can you foresee any issues coming from the other affected lines of code? Specifically, the following:
```
                    wooCommerceStore.fetchWooCommerceSite(it)
                    wooCommerceStore.fetchSiteGeneralSettings(it)
                    wooCommerceStore.fetchSiteProductSettings(it)
```

Also, I'm wondering about benefits of putting this behind a feature flag instead of simply removing the code. It seems the feature flag will require someone to remove the code in the future. Have you considered removing the code and using `git` in case we need to revert the change in the future?


> There is no practical way to test it. Just verify that the task won't be invoked anymore.

It's not exactly the test we are looking for, but what I think we might want to try is to put the app into background and disable WooCommerce on the site => bring the app to foreground and observe if the app just shows errors (expected) or crashes (not-expected)? Wdyt?

samiuelson committed:
Revert "Disable `updateSelectedSite` using feature flag"

This reverts commit ef0eb9f132e4132812950d33976e79b1e494df7c.

samiuelson committed:
Stop resetting site data when resuming an app

samiuelson commented:
Thanks, Jirka!

> It seems this PR removes all updates of the site when the app comes from background. Can you foresee any issues coming from the other affected lines of code? Specifically, the following:
> 
> ```
>                     wooCommerceStore.fetchWooCommerceSite(it)
>                     wooCommerceStore.fetchSiteGeneralSettings(it)
>                     wooCommerceStore.fetchSiteProductSettings(it)
> ```
> 
Good catch! I've reverted the initial changes (692c6fd) and removed just the line resetting the site data (479578d).

> Also, I'm wondering about benefits of putting this behind a feature flag instead of simply removing the code. It seems the feature flag will require someone to remove the code in the future. Have you considered removing the code and using `git` in case we need to revert the change in the future?

Yes, I agree it's better to remove the code.

> > There is no practical way to test it. Just verify that the task won't be invoked anymore.
> 
> It's not exactly the test we are looking for, but what I think we might want to try is to put the app into background and disable WooCommerce on the site => bring the app to foreground and observe if the app just shows errors (expected) or crashes (not-expected)? Wdyt?

Makes sense. I've updated the testing instructions.

samiuelson committed:
Clean up code
